### PR TITLE
Feature/[ENG-1953] admin app update registration schemas

### PR DESCRIPTION
**Pull request by jwalz**
## Purpose
ENG-1953:
Adding a Management Command to the Admin App to migrate Registration Schemas/Schemablocks on demand

## Changes
Added the new management command and updated the schema-related functions in migrations.py to better support non-migration callers.

Also made a change in website/projects/metadata/schemas.py to read schema files on-demand instead of only when the server is loaded. Without this change, the Admin App needed to be restarted before running the command in order to pick up any local changes to schemas since the last restart.

## QA Notes

  - Does this change require a data migration? If so, what data will we migrate? no
  - What is the level of risk?
    - Any permissions code touched? no
    - Is this an additive or subtractive change, other? additive
  - How can QA verify? (Through UI, API, AdminApp or AdminAdminApp?) AdminApp
    - If verifying through API, what's the new version? Please include the endpoints in PR notes or Dev docs.
  - What features or workflows might this change impact?
  - How will this impact performance?
-->

## Ticket
https://openscience.atlassian.net/browse/ENG-1953

### CODE REVIEW

File: osf/utils/migrations.py [hunk 1]
```diff
@@ -159,17 +160,16 @@ def remove_licenses(*args):
...
-            RegistrationSchema = args[0].get_model('osf', 'metaschema')
-        except Exception:
-            # Working outside a migration
-            from osf.models import RegistrationSchema
-    for schema in OSF_META_SCHEMAS:
-        schema_obj, created = RegistrationSchema.objects.update_or_create(
+        schema_model = state.get_model('osf', 'registrationschema')
+    except LookupError:
+        # Use MetaSchema model if migrating from a version before RegistrationSchema existed
+        schema_model = state.get_model('osf', 'metaschema')
```
Comment (Johnetordoff) [inline osf/utils/migrations.py:169]: Drive-by: I'm 90% sure we can totally delete this now.

File: osf/utils/migrations.py [hunk 2]
```diff
@@ -197,9 +197,10 @@ def create_schema_block(state, schema_id, block_type, display_text='', required=
     """
     For mapping schemas to schema blocks: creates a given block from the specified parameters
     """
-    RegistrationSchemaBlock = state.get_model('osf', 'registrationschemablock')
+    state = state or apps
+    schema_block_model = state.get_model('osf', 'registrationschemablock')
```
Comment (Johnetordoff) [inline osf/utils/migrations.py:201]: Drive-by: A little bit of a strange situation because we getting the model in different way, but model definitions always follow camel case.

**Comment by jwalz** [comment]
changes incorporated into https://github.com/CenterForOpenScience/osf.io/pull/9649
